sdk/servicebus/Azure.Messaging.ServiceBus/tests/RuleManager/RuleManagerLiveTests.cs

```diff
@@ -0,0 +1,588 @@
+
+                ServiceBusRuleManager ruleManager = client.CreateRuleManager(scope.TopicName, scope.SubscriptionNames.First());
+                IEnumerable<RuleDescription> rulesDescription;
+
+                rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual(RuleDescription.DefaultRuleName, rulesDescription.First().Name);
+
+                await ruleManager.RemoveRuleAsync(RuleDescription.DefaultRuleName);
+
+                await ruleManager.AddRuleAsync(new RuleDescription
+                {
+                    Filter = new TrueFilter(),
+                    Name = "BooleanFilter"
+                });
+
+                rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual("BooleanFilter", rulesDescription.First().Name);
+
+                await SendMessages(client, scope.TopicName);
+
+                var expectedOrders = Orders.ToList();
+                await ReceiveMessages(
+                    client,
+                    scope.TopicName,
+                    scope.SubscriptionNames.First(),
+                    expectedOrders);
+            }
+        }
+
+        /// <summary>
+        /// False boolean filter does not receive any messages.
+        /// </summary>
+        [Test]
+        public async Task FalseFilter()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithTopic(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+
+                ServiceBusRuleManager ruleManager = client.CreateRuleManager(scope.TopicName, scope.SubscriptionNames.First());
+
+                IEnumerable<RuleDescription> rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual(RuleDescription.DefaultRuleName, rulesDescription.First().Name);
+
+                await ruleManager.RemoveRuleAsync(RuleDescription.DefaultRuleName);
+
+                await ruleManager.AddRuleAsync(new RuleDescription
+                {
+                    Filter = new FalseFilter(),
+                    Name = "BooleanFilter"
+                });
+
+                rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual("BooleanFilter", rulesDescription.First().Name);
+
+                await SendMessages(client, scope.TopicName);
+
+                var receiver = client.CreateReceiver(scope.TopicName, scope.SubscriptionNames.First());
+                var messages = await receiver.ReceiveBatchAsync(Orders.Length, TimeSpan.FromSeconds(10));
+                Assert.True(messages.Count() == 0);
+            }
+        }
+
```

Comment:
Can we add a test involving transactions? It should be possible to do rule operations and other settlement/send stuff as part of the same transaction. The use of rule operations in transactions was one of the main reasons for including the rule management as part of the AMQP client.
Probably better to put this in TransactionLiveTests.

Reply:
Added!

Reply:
Any chance we can include a version of the test where we don't call ts.Complete()? This will ensure that the rule management operation is being included in the transaction as we expect that it would get rolled back.

Reply:
If I don't call ts.Complete() at the end of the transaction to commit, it is not rolling back the transaction as expected. I also checked the same behaviour in Track-1.

For now I've commented out the transaction rule management test cases. We will uncomment/remove it later once we get the clarification from the Service team.